### PR TITLE
CMR-9802: Clean up common-lib removing unnecessary util usage

### PR DESCRIPTION
**Pull request by TylerHeald1**
# Overview
Some clean-up of common-lib
### What is the feature/fix?

Some namespaces were identified for removal or moving to a more appropriate space, so that was done

### What is the Solution?

cmr.common.id-generator removed
cmr.common.regex-builder deleted, the only usage of it was swapped for regex literals
cmr.common-app.services.ingest.subscription-common moved to common.util

### What areas of the application does this impact?

common lib

# Checklist

- [x] I have updated/added unit and integration tests that prove my fix is effective or that my feature works
- [x] New and existing unit and int tests pass locally and remotely with my changes
- [ ] I have removed unnecessary/dead code and imports in files I have changed
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have cleaned up integration tests by doing one or more of the following:
  - de-duped, consolidated, removed dead int tests
  - transformed applicable int tests into unit tests
  - refactored to reduce number of system state resets by updating fixtures (use-fixtures :each (ingest/reset-fixture {})) to be :once instead of :each

### CODE REVIEW

File: common-app-lib/src/cmr/common_app/services/ingest/subscription_common.clj [hunk 1]
```diff
@@ -1,20 +0,0 @@
-(ns cmr.common-app.services.ingest.subscription-common
```
Comment (jmaeng72) [inline common-app-lib/src/cmr/common_app/services/ingest/subscription_common.clj:1]: unused?

File: common-lib/test/cmr/common/test/util.clj [hunk 2]
```diff
@@ -972,3 +972,21 @@
 
     "nil as input"
     nil nil))
+
+
+(deftest normalize-parameters-test
```
Comment (jmaeng72) [inline common-lib/test/cmr/common/test/util.clj:977]: I think you can use are3 here

File: common-app-lib/src/cmr/common_app/services/ingest/subscription_common.clj (hunk 1 above)
Comment (TylerHeald1) [reply common-app-lib/src/cmr/common_app/services/ingest/subscription_common.clj:1]: cmr.common-app.services.ingest.subscription-common moved to common.util

File: common-lib/test/cmr/common/test/util.clj (hunk 2 above)
Comment (TylerHeald1) [reply common-lib/test/cmr/common/test/util.clj:977]: Swapped to are3

**Review by jmaeng72 — APPROVED** [review]

**Review by eereiter — APPROVED** [review]

**Pull request merged**